File: solutions/scrabble-score/PHP/ScrabbleScore.php
<?php

declare(strict_types=1);

const SCORES = [
    'AEIOULNRST' => 1,
    'DG' => 2,
    'BCMP' => 3,
    'FHVWY' => 4,
    'K' => 5,
    'JX' => 8,
    'QZ' => 10
];

function score(string $word): int
{
    $total = 0;
    foreach ( count_chars(strtoupper($word), 1) as $i => $count ) {
        foreach (SCORES as $alphabets => $score) {
            if (strpos($alphabets, chr($i)) !== false) {
                $total += $count * $score;
                break;
            }
        }
    }

    return $total;
}

File: solutions/scrabble-score/Zig/scrabble_score.zig
pub fn score(s: []const u8) u32 {
    var total: u32 = 0;
    for (s) |c| {
        const letter: u32 = switch (c) {
            'a', 'A', 'e', 'E', 'i', 'I', 'o', 'O', 'u', 'U', 'l', 'L', 'n', 'N', 'r', 'R', 's', 'S', 't', 'T' => 1,
            'd', 'D', 'g', 'G' => 2,
            'b', 'B', 'c', 'C', 'm', 'M', 'p', 'P' => 3,
            'f', 'F', 'h', 'H', 'v', 'V', 'w', 'W', 'y', 'Y' => 4,
            'k', 'K' => 5,
            'j', 'J', 'x', 'X' => 8,
            'q', 'Q', 'z', 'Z' => 10,
            else => 0,
        };
        total += letter;
    }
    return total;
}
